pkg/asset/machines/aws/zones.go

```diff
@@ -8,8 +8,16 @@ import (
 	awsutil "github.com/openshift/installer/pkg/asset/installconfig/aws"
 )
 
+var cache map[string][]string
```

Comment:
I would prefer to see this as the following.

```
var cache = map[string][]string{}
```

It creates the map when it may never be used. But I don't think that is a terrible cost to pay to avoid having to check for nil in `AvailabilityZones`. If you want to wait to create the map, then the creation should be down right before the insertion into the map instead of near the read of the map. Go lets you read from a nil map.